test/smokes/hadolint/config_option/sider.yml

```
@@ -0,0 +1,3 @@
+linter:
+  hadolint:
+    config: .hadolint.yml
```

[nitpick] Thank you for your adding the test case!

`.hadolint.yml` is **not** supported by default (correct is `.hadolint.yml`), so this file name is a bit confusing. 😅 

How about renaming like this?

```suggestion
    config: custom-hadolint.yml
```

See https://github.com/hadolint/hadolint#configure